File: ms-productos/src/main/java/com/ferreteria/productos/service/ProveedorService.java
package com.ferreteria.productos.service;

import com.ferreteria.productos.dto.ProveedorRequest;
import com.ferreteria.productos.dto.ProveedorResponse;
import java.util.List;

public interface ProveedorService {
    List<ProveedorResponse> listar();
    ProveedorResponse obtenerPorId(Long id);
    ProveedorResponse crear(ProveedorRequest request);
    ProveedorResponse actualizar(Long id, ProveedorRequest request);
    void eliminar(Long id);
}

File: frontend-angular/src/app/services/proveedor.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Proveedor } from '../models/proveedor.model';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ProveedorService {
  private readonly apiUrl = `${environment.apiUrl}/proveedores`;

  constructor(private http: HttpClient) {}

  listar(): Observable<Proveedor[]> {
    return this.http.get<Proveedor[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Proveedor> {
    return this.http.get<Proveedor>(`${this.apiUrl}/${id}`);
  }

  crear(proveedor: Omit<Proveedor, 'idProveedor'>): Observable<Proveedor> {
    return this.http.post<Proveedor>(this.apiUrl, proveedor);
  }

  actualizar(proveedor: Proveedor): Observable<Proveedor> {
    return this.http.put<Proveedor>(`${this.apiUrl}/${proveedor.idProveedor}`, proveedor);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
